diff --git a/command.js b/command.js
index 586c774f717ade45a01b7b9dd699a77a0e13f22d..ab164f152e3bff74fcfa1fa0c4502e8df977a35a 100644
--- a/command.js
+++ b/command.js
@@ -166,6 +166,10 @@ exports.builder = (yargs) => {
       /* eslint-enable no-param-reassign */
 
       return argv;
+    }).option("rw-custom-bump", {
+      hidden: true,
+      type: "boolean",
+      default: false,
     });
 };
 
diff --git a/index.js b/index.js
index 716c8764f501af5fe0e83fc38d6dfa2b994877a8..e4b24e79c2cc78e14a976ba9f07087673f5236e2 100644
--- a/index.js
+++ b/index.js
@@ -424,7 +424,11 @@ class PublishCommand extends Command {
       (fallback) =>
       ({ lastVersion = fallback, refCount, sha }) => {
         // the next version is bumped without concern for preid or current index
-        const nextVersion = semver.inc(lastVersion.replace(this.tagPrefix, ""), release.replace("pre", ""));
+        let nextVersion = semver.inc(lastVersion.replace(this.tagPrefix, ""), release.replace("pre", ""));
+
+        if (this.argv.rwCustomBump) {
+          nextVersion = semver.inc(nextVersion, "minor")
+        }
 
         // semver.inc() starts a new prerelease at .0, git describe starts at .1
         // and build metadata is always ignored when comparing dependency ranges
