diff --git a/dist/repo/github.js b/dist/repo/github.js
index c9ac17f256c6b3529f386bc4c727ae4521256c76..1dde5dd36db340cbfbf029539c73874aaf9b9eeb 100644
--- a/dist/repo/github.js
+++ b/dist/repo/github.js
@@ -58,7 +58,7 @@ function getNextLink(link) {
     return null;
   }
 
-  return nextLink.split(';')[0].slice(1, -1);
+  return nextLink.split(';')[0].trim().slice(1, -1);
 }
 
 function getContributorsPage(githubUrl, optionalPrivateToken) {
