diff --git a/dist/node/chunks/dep-df561101.js b/dist/node/chunks/dep-df561101.js
index 1bc8674177fe73120171b22436e6104713c5d764..f0fee7b385868cb01c6d47b80d7f64a7368c0412 100644
--- a/dist/node/chunks/dep-df561101.js
+++ b/dist/node/chunks/dep-df561101.js
@@ -55890,12 +55890,12 @@ async function instantiateModule(url, server, context = { global }, urlStack = [
     };
     urlStack = urlStack.concat(url);
     const isCircular = (url) => urlStack.includes(url);
-    const { isProduction, resolve: { dedupe, preserveSymlinks }, root, } = server.config;
+    const { isProduction, resolve: { dedupe, preserveSymlinks, conditions }, root, } = server.config;
     const resolveOptions = {
         mainFields: ['main'],
         browserField: true,
         conditions: [],
-        overrideConditions: ['production', 'development'],
+        overrideConditions: [...conditions, 'production', 'development'],
         extensions: ['.js', '.cjs', '.json'],
         dedupe,
         preserveSymlinks,
