diff --git a/react/hooks/internal/useRenderGuard.js b/react/hooks/internal/useRenderGuard.js
index 15551e6e2de4dcc585e9426f80c67cb136ecd2a7..bf7ee0216f2b18b1613454b46b620adbf204cffa 100644
--- a/react/hooks/internal/useRenderGuard.js
+++ b/react/hooks/internal/useRenderGuard.js
@@ -1,7 +1,9 @@
 import * as React from "rehackt";
 function getRenderDispatcher() {
-    var _a, _b;
-    return (_b = (_a = React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED) === null || _a === void 0 ? void 0 : _a.ReactCurrentDispatcher) === null || _b === void 0 ? void 0 : _b.current;
+    var r18 = typeof React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED !== "undefined" ? React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED : {};
+    var r19 = typeof React.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE !== "undefined" ? React.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE : {};
+    var _b = r18.ReactCurrentDispatcher || r19.ReactCurrentDispatcher;
+    return _b === null || _b === void 0 ? void 0 : _b.current;
 }
 var RenderDispatcher = null;
 /*
diff --git a/testing/internal/profile/profile.js b/testing/internal/profile/profile.js
index bcb709e4a7f6a72b381632e6da6a5c7d9ab5d8f1..953121f90df66e2dd5a6276a710d9ff874719660 100644
--- a/testing/internal/profile/profile.js
+++ b/testing/internal/profile/profile.js
@@ -276,8 +276,11 @@ export function profileHook(renderCallback) {
     });
 }
 function resolveHookOwner() {
-    var _a, _b, _c;
-    return (_c = (_b = (_a = React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED) === null || _a === void 0 ? void 0 : _a.ReactCurrentOwner) === null || _b === void 0 ? void 0 : _b.current) === null || _c === void 0 ? void 0 : _c.elementType;
+    var r18 = typeof React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED !== "undefined" ? React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED : {};
+    var r19 = typeof React.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE !== "undefined" ? React.__CLIENT_INTERNALS_DO_NOT_USE_OR_WARN_USERS_THEY_CANNOT_UPGRADE : {};
+    var _b = r18.ReactCurrentOwner || r19.ReactCurrentOwner;
+    var _c;
+    return (_c = _b === null || _b === void 0 ? void 0 : _b.current) === null || _c === void 0 ? void 0 : _c.elementType;
 }
 export function useTrackRenders(_a) {
     var _b = _a === void 0 ? {} : _a, name = _b.name;
